chore(website): streamlined Getting Started docs (#5248)
typescript-eslint/typescript-eslint: docs/linting/TYPED_LINTING.md

- `parserOptions.tsconfigRootDir` tells our parser the absolute path of your project's root directory.
- `parserOptions.project` tells our parser the relative path where your project's `tsconfig.json` is.
  - If your project is a multi-package monorepo, see [our docs on configuring a monorepo](./typed-linting/MONOREPOS.md).
- `plugin:@typescript-eslint/recommended-requiring-type-checking` is another recommended configuration we provide. This one contains rules that specifically require type information.

With that done, run the same lint command you ran before.
You may see new rules reporting errors based on type information!

## FAQs

### How is performance?

Typed rules come with a catch.
By including `parserOptions.project` in your config, you incur the performance penalty of asking TypeScript to do a build of your project before ESLint can do its linting.
For small projects this takes a negligible amount of time (a few seconds or less); for large projects, it can take longer.

> Most of our users do not mind this cost, as it is negligibly small, or failing that worth the power of the more powerful type-aware static analysis rules.

Reviewer: I probably wouldn't say "negligibly small" here just because it can quickly scale out of that range.

```suggestion
Most of our users do not mind this cost as the power and safety of type-aware static analysis rules is worth the tradeoff.
```
